docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,394 @@
 Autograd mechanics
 ==================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note will present an overview of how autograd works. It's not strictly necessary
+to understand all this, but we recommend getting familiar with it, as it will help you
+write more efficient, cleaner programs, and can aid you in debugging.
 
-.. _excluding-subgraphs:
+TL-DR:
```

Comment:
Since you are summarizing each section, how about calling this "Summary" and using bold for the name of each section?

Reply:
just a thought: would "table of content" be more fitting?

Reply:
Or at the very least, use the right formatting: TL;DR ;)